spec/providers/coreIntegration/rtcpeerconnection.integration.src.js

```diff
@@ -33,23 +31,35 @@ module.exports = function (pc, dc, setup) {
     bob.on('onicecandidate', function (msg) {
       msg.candidate && alice.addIceCandidate(msg.candidate);
     });
+    alice.on('onicecandidate', function (msg) {
+      if (!msg.candidate) {
+        bob.setRemoteDescription(aliceCandidates.shift()).then(function () {
+          var last;
+          while (aliceCandidates.length) {
+            last = bob.addIceCandidate(aliceCandidates.shift())
+          }
+          return last;
+        }).then(function () {
+          return bob.createAnswer();
+        }).then(function (answer) {
+          return bob.setLocalDescription(answer).then(function () {return answer; });
+        }).then(function (answer) {
+          return alice.setRemoteDescription(answer);
+        }, function (err) {
+          console.error('RTC failed: ',err);
+        });
+      } else {
+        aliceCandidates.push(msg.candidate);
+      }
+    });
     alice.createDataChannel('channel').then(function (id) {
       aliceChannel = datachan(id);
       aliceChannel.on('onopen', function () {
         aliceChannel.send('message from alice');
       });
       alice.createOffer().then(function (offer) {
-        return alice.setLocalDescription(offer).then(function () {return offer; });
-      }).then(function (offer) {
-        return bob.setRemoteDescription(offer);
-      }).then(function () {
-        return bob.createAnswer();
-      }).then(function (answer) {
-        return bob.setLocalDescription(answer).then(function () {return answer; });
-      }).then(function (answer) {
-        return alice.setRemoteDescription(answer);
-      }, function (err) {
-        console.error('RTC failed: ',err);
+        alice.setLocalDescription(offer);
+        aliceCandidates.push(offer);
```

Comment:
This seems pretty confusing to me.  I'd rather have a separate `aliceOffer` variable.